inst/stan/functions/obs_lpmf.stan

```diff
@@ -21,14 +22,25 @@ real obs_lpmf(int[] dummy, int start, int end, int[] obs, int[] sl, int[] csl,
     tar_obs = imp_obs[g][t];
     // allocate report day effects
     rdlh = srdlh[rdlurd[t:(t + l - 1), g]];
+    // allocate share of cases with known reference date
+    tar_alpha = alpha[g][t];
     // combine expected final obs and date effects to get expected obs
     esnap += l;
-    exp_obs[ssnap:esnap] = expected_obs(
+    vector[l] exp_obs_all = expected_obs(
       tar_obs, ref_lh[1:l, dpmfs[i]], rdlh, ref_p
     );
+    // compute expected final obs with known and missing reference date
+    exp_obs[ssnap:esnap] = exp_obs_all * tar_alpha;
+    if(t+l-1>=1+dmax){
+      exp_obs_miss[g][max(1 + dmax, t):(t + l - 1)] += exp_obs_all[max(2 + dmax - t, 1):l] * (1 - tar_alpha);
+    }
     ssnap += l;
   }
-  // observation error model (across all reference times and groups)
-  tar = neg_binomial_2_lupmf(snap_obs | exp_obs, phi);
+  // observation error model with known reference dates (across all reference times and groups)
+  tar = neg_binomial_2_lupmf(obs | exp_obs, phi);
+  // observation error model with missing reference dates
+  for (k in 1:n_groups) {
+    tar += neg_binomial_2_lupmf(obs_miss[k][(1 + dmax):n_obs_miss] | exp_obs_miss[k][(1 + dmax):n_obs_miss], phi);
+  }
```

Comment:
As can be seen, I did not go for a flattening of the `obs_miss` for now. Should discuss if we want this here as well.

Reply:
😡 😆